projects/ion/src/lib/utils/generateID.ts

```diff
@@ -0,0 +1,17 @@
+const MAX_LENGTH = 9999;
+
+export const COOLDOWN_TIME = 400;
+
+export const generateIDs = (prefix: string, testeid: string): string => {
```

Comment:
I believe this function can be replaced by this one:

```suggestion
export const generateIDs = (prefix: string, testeid: string): string => {
  let id = 1;
  document
    .querySelectorAll(`*[data-testid="${testeid}"]`)
    .forEach((element) => {
      const startID = prefix.length;
      const elementID = parseInt(element.id.substring(startID, MAX_LENGTH));
      id = elementID >= id ? elementID + 1 : id;
    });
  return prefix + id;
};
```

In my opinion, it is more optimized because it avoids the creation of a new array to store the elements found. The forEach method iterates over each element directly in the NodeList returned by querySelectorAll, which is more memory efficient than creating a new array with Array.from.